### DISCUSSION TITLE
HTTP Outbound Proxy Support

### DISCUSSION THREAD

**Discussion by ThomasCr** [nextcloud/all-in-one#3830] [Ideas] [answered]
Hi,

I tested to upgrade from the ["normal" docker container setup](https://hub.docker.com/_/nextcloud) to the aio setup but cant manage to start the master container "nextcloud-aio-nextcloud" with the follow error in the logfile:

```
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: Permission denied
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: Permission denied
ERROR: unable to select packages:
  imagemagick (no such package):
    required by: world[imagemagick]
The packet imagemagick was not installed!
Enabling Imagick...
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: Permission denied
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: Permission denied
ERROR: unable to select packages:
  .docker-php-ext-enable-deps-20231127.124346:
    masked in: cache
    satisfies: world[.docker-php-ext-enable-deps=20231127.124346]
  binutils (no such package):
    required by: .docker-php-ext-enable-deps-20231127.124346[binutils]
Could not install PHP extension imagick!
Configuring Redis as session handler...
Checking connection to appstore
Curl didn't produce a 200 status, is appstore reachable?
```

That's because, we have strict policy to only allow outbound traffic with our http proxy server (and filter there). Before (with the old container setup) I was able to setup the environment variables by myself and added this:

```
    environment:
      - http_proxy=${HTTP_PROXY}
      - https_proxy=${HTTPS_PROXY}
      - no_proxy=${NO_PROXY}
```
So I got internet access on the containers. But now it is not possible to set the proxy server. It would be nice, to have a Input mask on the AIO Interface to enter those details.

**Comment by szaimen** [comment, marked as answer]
Hi, currently there are no plans to implement such a feature. 

However you might be able to make it work with https://github.com/nextcloud/all-in-one/tree/main/manual-install

**Comment by ozon2** [comment]
For anyone looking to make the proxy work without the manual install, I found this solution works for me:
https://help.nextcloud.com/t/aio-with-environmental-variables-for-derived-containers/175836/12

>I could fix it by editing the nextcloud_aio_nextcloud/_data/config/config.php and adding the line:
>
> 'proxy' => 'http://125.1.1.xxx:3128',
>
> After restarting the docker service all the containers are working fine

**Reply by yingziwu** [reply]
thanks, it work for me.

**Comment by yeison** [comment]
I am getting the same exactly the same issues after upgrading to the latest version of nextcloud.  I thought it was reverse-proxy related but it's not.  I tried it without my reverse proxy and still had the exact same issue.  After spending two days to realize I had to revert to an older version, I realized that something is broken in the most recent version that is causing this error:

```
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: Permission denied
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: Permission denied
ERROR: unable to select packages:
  imagemagick (no such package):
    required by: world[imagemagick]
The packet imagemagick was not installed!
Enabling Imagick...
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: Permission denied
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: Permission denied
ERROR: unable to select packages:
  .docker-php-ext-enable-deps-20231127.124346:
    masked in: cache
    satisfies: world[.docker-php-ext-enable-deps=20231127.124346]
  binutils (no such package):
    required by: .docker-php-ext-enable-deps-20231127.124346[binutils]
Could not install PHP extension imagick!
Configuring Redis as session handler...
Checking connection to appstore
Curl didn't produce a 200 status, is appstore reachable?
```

The most recent versions of the software get stuck in an infinite loop trying to reach app store and failing. The old versions do not have the curl error:

```
2025-05-06T11:05:14.889788323Z WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.21/main: temporary error (try again later)
2025-05-06T11:05:19.893861726Z WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.21/community: temporary error (try again later)
2025-05-06T11:05:19.898712382Z ERROR: unable to select packages:
2025-05-06T11:05:19.898722081Z   .docker-php-ext-enable-deps-20250506.110519:
2025-05-06T11:05:19.898725096Z     masked in: cache
2025-05-06T11:05:19.898727601Z     satisfies: world[.docker-php-ext-enable-deps=20250506.110519]
2025-05-06T11:05:19.898730025Z   binutils (no such package):
2025-05-06T11:05:19.898732330Z     required by: .docker-php-ext-enable-deps-20250506.110519[binutils]
2025-05-06T11:05:19.901678408Z Could not install PHP extension imagick!
2025-05-06T11:05:19.906114588Z Connection to nextcloud-aio-redis (172.23.0.4) 6379 port [tcp/redis] succeeded!
2025-05-06T11:05:20.707775962Z Applying one-click-instance settings...
```

I downgraded to version 10.5.0 for anyone in the internet having this same issue:

docker image: nextcloud/all-in-one:20250114_092611
